src/rust/src/x509/verify.rs

```diff
@@ -366,21 +366,19 @@ fn build_subject_owner(
     py: pyo3::Python<'_>,
     subject: &pyo3::Py<pyo3::PyAny>,
 ) -> pyo3::PyResult<SubjectOwner> {
-    let subject = subject.as_ref(py);
+    let subject = subject.bind(py);
```

Comment:
If you add a `clone()` here, I don't believe all the temporary variables are required below.

Reply:
I added:
```rust
    let subject = subject.bind(py).clone();
```
But I still get:

```
    |
372 |           let value = subject
    |  _____________________^
373 | |             .getattr(pyo3::intern!(py, "value"))?
    | |_________________________________________________^ creates a temporary value which is freed while still in use
374 |               .downcast::<pyo3::types::PyString>()?;
    |                                                    - temporary value is freed at the end of this statement
375 |
376 |           Ok(SubjectOwner::DNSName(value.to_str()?.to_owned()))
    |                                    ----- borrow later used here
    |
    = note: consider using a `let` binding to create a longer lived value
```

Reply:
Oh. Put a clone at the end of that expression. It's more readable than the intermediate bounds.